src/main/java/com/tenx/ms/retail/client/service/ClientService.java

```diff
@@ -0,0 +1,69 @@
+package com.tenx.ms.retail.client.service;
+
+import com.tenx.ms.retail.client.domain.ClientEntity;
+import com.tenx.ms.retail.client.repository.ClientRepository;
+import com.tenx.ms.retail.client.rest.dto.Client;
+import com.tenx.ms.retail.client.util.ClientConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Created by goropeza on 30/08/16.
+ */
+@Service
+public class ClientService {
+    @Autowired
+    private ClientRepository clientRepository;
+    @Autowired
+    private ClientConverter clientConverter;
+
+    public Client findClientById(Long clientId) {
+        Optional<ClientEntity> result = clientRepository.findById(clientId);
+        if (result.isPresent()) {
+            return clientConverter.repositoryToApiModel(result.get());
+        } else {
+            throw new NoSuchElementException();
```

Comment:
It's better to throw an exception that includes an error message instead of simply throwing the exception.